src/parser/TTML.cpp

```diff
@@ -290,11 +291,14 @@ uint64_t TTML2SRT::GetTime(const char *tmchar)
   else
   {
     unsigned int th, tm, ts, tms;
-    char del;
-    if (sscanf(tmchar, "%u:%u:%u%c%2u", &th, &tm, &ts, &del, &tms) == 5)
+    char del, ctms[3];
+    if (sscanf(tmchar, "%u:%u:%u%c%3c", &th, &tm, &ts, &del, ctms) == 5)
     {
+      sscanf(ctms, "%3u", &tms);
+      if (strlen(ctms) == 2)
+        tms = tms*10;
       ret = th * 3600 + tm * 60 + ts;
-      ret = ret * 1000 + tms * 10;
+      ret = ret * 1000 + tms;
```

Comment:
If you have 429330:07:12.359 and reading 2 should lead to 35.
35*10 = 350, you'll loose 9ms, which is half of a 60fps frame rendered.

You are really sure that 9ms will be visible for anyone? Even it is not a ful frame duration?
I doubt, really.

Reply:
The difference of 9 ms is not really the problem, but when there are 3 digits and the number is not a multiple of 10, the subtitles are simply not shown in Kodi.

So if you convert 359 ms to 350 ms so the GetTime-function returns 350 ms, Kodi doesn't show the subtitles at all. If the GetTime-function returns 359 ms, the subtitles are totally shown.

Reply:
I investigated this problem a bit further and came to the conclusion that to loss of 9 ms precision interferes with the following piece of code in the StackSubTitle-function:
```c++
  if (sub.start < m_ptsOffset)
  {
    sub.start += m_ptsOffset;
    sub.end += m_ptsOffset;
  }
```
When  the GetTime-function returns 350 ms instead of 359 ms, then the sub.start is less than the m_ptsOffset and the subtitle is not displayed.

So, my tentative conclusion is that the maintaining the subtitle timestamp precision in milliseconds is necessary to avoid unwanted effects in other parts of the code.

Could you please let me know what your opinion on this problem is.

Is my fix ready to be merged?

Reply:
Does this correct the problem of subtitles with carriage return?
Some users have reported this problem with Netflix plugin see screenshots:
![image](https://user-images.githubusercontent.com/3257156/52535178-978e6700-2d4b-11e9-8836-dc384ffbd41c.png)
the subtitle positioned at the top-left should be positioned below last

Reply:
Maybe, my commit fixes **three** bugs.